docs/content/preview/explore/ysql-language-features/join-strategies.md

```diff
@@ -0,0 +1,256 @@
+               Storage Table Read Execution Time: 1001.250 ms
+ Planning Time: 0.252 ms
+ Execution Time: 2003.039 ms
+ Storage Read Requests: 2
+ Storage Read Execution Time: 2002.493 ms
+ Storage Execution Time: 2002.493 ms
+ Peak Memory Usage: 126 kB
+```
+
+## Hash Join
+
+The Hash Join algorithm starts by hashing the join key columns of both input tables. The hash function transforms the join key values into a hash code. Hash Joins are suitable for equi-joins (joins based on equality) and are effective when the join keys are not sorted or indexed.
+
+To fetch all scores of students named `Natasha` who have scored more than `70` in any subject using Hash join, you would execute,
+
+```sql
+explain (analyze, dist, costs off)
+/*+
+    hashjoin(students scores)
+    set(yb_enable_optimizer_statistics on)
+*/
+SELECT name, subject, score 
+      FROM students JOIN scores USING(id) 
+      WHERE name = 'Natasha' and score > 70;
+```
+
+The query plan would be similar to:
+
+```sql
+ Hash Join (actual time=2002.431..2002.477 rows=18 loops=1)
+   Hash Cond: (scores.id = students.id)
+   ->  Seq Scan on scores (actual time=1001.043..1001.061 rows=51 loops=1)
+         Remote Filter: (score > 70)
+         Storage Table Read Requests: 1
+         Storage Table Read Execution Time: 1000.927 ms
+   ->  Hash (actual time=1001.360..1001.361 rows=8 loops=1)
+         Buckets: 1024  Batches: 1  Memory Usage: 9kB
+         ->  Seq Scan on students (actual time=1001.327..1001.335 rows=8 loops=1)
+               Remote Filter: ((name)::text = 'Natasha'::text)
+               Storage Table Read Requests: 1
+               Storage Table Read Execution Time: 1001.211 ms
+ Planning Time: 0.231 ms
+ Execution Time: 2002.540 ms
+ Storage Read Requests: 2
+ Storage Read Execution Time: 2002.137 ms
+ Storage Execution Time: 2002.137 ms
+ Peak Memory Usage: 141 kB
+```
+
+## Batched Nested Loop Join (BNL)
+
+In the case of the Nested loop joins, the inner table is accessed multiple times one for each outer table row. This would lead to multiple RPC requests across the different nodes in the cluster, making this join strategy very slow as the outer table gets larger.
+
+To reduce the number of requests sent across the nodes during the Nested Loop join, YugabyteDB adds a new optimization to batch multiple keys of the outer table into one RPC request. This batch size can be controlled using the GUC variable `yb_bnl_batch_size`, which defaults to `1` (which effectively means that the feature is `OFF`). The suggested value for this variable is `1024`.
+
+If `yb_bnl_batch_size` is greater than `1`, the optimizer will try to adopt the batching optimization when other join strategies are not fit for the current query.
+
+To fetch all scores of students named `Natasha` who have scored more than `70` in any subject using Batched nested loop join, you would execute,
+
+```sql
+explain (analyze, dist, costs off)
+/*+
+    nestloop(students scores)
```

Comment:
Remember that this hint forces an NL join in 2.21+.

Reply:
Ok . will make a note to update after 2.21 release.

Reply:
done.